src/components/LanguageSelector/LanguageSelector.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+import { Menu } from '../header/Menu/Menu'
+import { LanguageSelectorButton } from './LanguageSelectorButton'
+import { Button } from '../Button/Button'
+
+export type LanguageDefinition = {
+  label: string
+  label_local?: string
+  attr: string
+  on_click: string | (() => void)
+}
+
+type LanguageSelectorProps = {
+  label?: string
+  langs: LanguageDefinition[]
+  small?: boolean
+  className?: string
+}
+
+export const LanguageSelector = ({
+  label,
+  langs,
+  small,
+  className,
+  ...divProps
+}: LanguageSelectorProps &
+  JSX.IntrinsicElements['div']): React.ReactElement => {
+  const classes = classnames(
+    'usa-language-container',
+    {
+      [`usa-language--small`]: small !== undefined,
+    },
+    className
+  )
+
+  if (langs.length > 2) {
+    const [isOpen, setIsOpen] = useState(false)
```

Comment:
Thought i'd jump the gun a bit and try this in IRS project. Getting the following eslint error there:
```
React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render.eslint[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)
```

Reply:
fixed